scrapy/code/groceries/groceries/spiders/safeway/scraper.py

```diff
@@ -0,0 +1,267 @@
+#! /usr/local/bin/python3
+
+import scrapy
+from scrapy.shell import inspect_response
+from scrapy_splash import SplashRequest
+from scrapy_selenium import SeleniumRequest
+
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from seleniumHelpers import create_parse_request, create_unfiltered_parse_request
+import time
+#TODO change to selenium so that we can update the location!!!
+import re
+
+from util import read_script, convert_cents, store_url, get_next_url, get_url_metadata, lookup_category, finish_url, get_next_pagination, trim_url, convert_to_ounces, convert_units, clean_string
+
+def convert_ppu(incoming_ppu):
+    if not incoming_ppu:
+        return ""
+    ppu = incoming_ppu
+    charactersToRemove = ['$', ' ']
+    for remove in charactersToRemove:
+        ppu = ppu.replace(remove,'')
+    if ppu.find('per') != -1:
+        ppuSplit = ppu.split('per')
+    elif ppu.find('each') != -1:
+        # Split off the each, and add it on manually
+        ppuSplit = ppu.split('each')
+        print ("ppu - " + ppu + ", ppuSplit - " + str(ppuSplit))
+        # Add 'each' as the units to be set appropriately
+        ppuSplit[1] = 'ea'
+
+    cost = ppuSplit[0]
+        # if theirs a / seperating multiple values
```

Comment:
Good comment. Could you change theirs to there is?

Reply:
Actually ended up removing the PPU section from safeway because it wasn't used.